### DISCUSSION TITLE
Adding languages to a fresh 1.2 installation

### DISCUSSION THREAD

**Discussion by gonzus** [LunarVim/LunarVim#3447] [Q&A]
I just backed-up my lvim installation and installed release 1.2 following the instructions -- everything works as expected.

I found that I cannot debug C++ programs; if I open a C++ file and type `<leader> d s`, I get the following error message:
```
No configuration found for `cpp`. You need to add configs to `dap.configurations.cpp`
```

Is the intention that adding TreeSitter and DAP support for a new language should be as easy as adding the language name to a list of languages? For example, `lvim.ts.languages = { "c++", "rust" }` and `lvim.dap.languages = { "c++", "rust" }` (or even a single list for both TS and DAP)?

If this is not the intention, then how can I add support for additional languages while keeping the ability to upgrade lvim, which is what I fear will happen if I start adding stuff everywhere?

**Comment by vonPB** [comment]
For now dap debuggers have to be set up manually. Fortunally we can use Mason to install the binaries for most of them.
Here is a sample config for cpp
https://github.com/LunarVim/starter.lvim/blob/c-ide/config.lua
